libs/ml/include/ml/ops/top_k.hpp

```diff
@@ -0,0 +1,200 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/tensor.hpp"
+#include "math/top_k.hpp"
+#include "ml/exceptions/exceptions.hpp"
+#include "ml/ops/ops.hpp"
+
+#include <cassert>
+#include <utility>
+#include <vector>
+
+namespace fetch {
+namespace ml {
+namespace ops {
+
+template <class T>
+class TopK : public fetch::ml::ops::Ops<T>
+{
+public:
+  using TensorType     = T;
+  using SizeType       = typename TensorType::SizeType;
```

Comment:
this is not allowed because it's shadowing.
you should do this:
using SizeType = fetch::math::SizeType;